index.js

```diff
@@ -343,6 +343,10 @@
     return xBool !== yBool ? or(x, y) : xEmpty;
   });
 
+  var wrapIndexOf = function(f) {
+    return R.curry(R.compose(R.ifElse(R.lte(0), Just, Nothing), f));
```

Comment:
Yes, but would it be more beautiful? :smile: But yeah, I agree. Actually would there be a way to make the whole function definition point free?